src/Nevermind/Nevermind.JsonRpc/JsonRpcService.cs

```diff
@@ -0,0 +1,207 @@
+            }
+            var items = new List<object>();
+            foreach (var item in collection)
+            {
+                var jsonItem = GetDataObject(item);
+                items.Add(jsonItem);
+            }
+            return GetSuccessResponse(items, request.Id);
+        }
+
+        private object GetDataObject(object data)
+        {
+            return data is IJsonRpcResult rpcResult ? rpcResult.ToJson() : data.ToString();
+        }
+
+        private object[] GetParameters(ParameterInfo[] expectedParameters, string[] providedParameters)
+        {
+            try
+            {
+                var executionParameters = new List<object>();
+                var i = 0;
+                foreach (var providedParameter in providedParameters)
+                {
+                    var expectedParameter = expectedParameters[i];
+                    var paramType = expectedParameter.ParameterType;
+                    object executionParam;
+                    if (typeof(IJsonRpcRequest).IsAssignableFrom(paramType))
+                    {
+                        executionParam = Activator.CreateInstance(paramType);
+                        ((IJsonRpcRequest)executionParam).FromJson(providedParameter);
+                    }
+                    else
+                    {
+                        executionParam = Convert.ChangeType(providedParameter, paramType);
+                    }
+                    executionParameters.Add(executionParam);
+                    i++;
+                }
+                return executionParameters.ToArray();
+            }
+            catch (Exception e)
+            {
+                _logger.Error("Error while parsing parameters", e);
+                return null;
+            }
+        }
+
+        private string GetSuccessResponse(object result, string id)
+        {
+            var response = new JsonRpcResponse
+            {
+                Jsonrpc = _configurationProvider.JsonRpcVersion,
+                Id = id,
+                Result = result
+            };
+            return _jsonSerializer.SerializeObject(response);
+        }
+
+        private string GetErrorResponse(ErrorType invalidRequest, string message, string id)
+        {
+            var response = new JsonRpcResponse
+            {
+                Jsonrpc = _configurationProvider.JsonRpcVersion,
+                Id = id,
+                Error = new Error
+                {
+                    Code = _configurationProvider.ErrorCodes[invalidRequest],
+                    Message = message
+                }
+            };
+            return _jsonSerializer.SerializeObject(response);
+        }
+
+        private (ErrorType?, string) Validate(JsonRpcRequest rpcRequest)
+        {
+            var methodName = rpcRequest.Method;
+            if (string.IsNullOrEmpty(methodName))
+            {
+                return (ErrorType.InvalidRequest, "Method is required");
```

Comment:
provide method name and request id in the log message so we can quickly find the cause of the error